src/Components/Web/src/HtmlRendering/StaticHtmlRenderer.HtmlWriting.cs

```diff
@@ -173,6 +178,29 @@ private int RenderElement(int componentId, TextWriter output, ArrayRange<RenderT
         }
     }
 
+    private int RenderScriptElementChildren(int componentId, TextWriter output, ArrayRange<RenderTreeFrame> frames, int position, int maxElements)
+    {
+        // Script tags are special. Unlike in XHTML, the contents of script elements in HTML
+        // are parsed as plain text, so HTML entities are not recognized and are treated as
+        // literal text. Code like console.log(&quot;Hello&quot;) would be a syntax error.
+        // HTML does not make use of <![CDATA[ ... ]]> either. So inside <script>...</script>,
+        // we must disable HTML encoding. This is hardly a XSS issue since if an application
+        // is rendering user content inside <script>, they already have an XSS problem unless
+        // they have carefully escaped the user content.
+        // https://stackoverflow.com/a/14781466
+        // https://html.spec.whatwg.org/multipage/scripting.html#restrictions-for-contents-of-script-elements
+        var originalEncoder = _htmlEncoder;
+        try
+        {
+            _htmlEncoder = _nullHtmlEncoder;
```

Comment:
There is a JavaScript encoder that we must use in these cases.

https://learn.microsoft.com/es-es/dotnet/api/system.text.encodings.web.javascriptencoder?view=net-7.0